packages/analytics-node/src/plugins/context.ts

```diff
@@ -0,0 +1,38 @@
+import { BeforePlugin, NodeConfig, Event, PluginType } from '@amplitude/analytics-types';
+import { UUID } from '@amplitude/analytics-core';
+import { VERSION } from '../version';
+
+export class Context implements BeforePlugin {
+  name = 'context';
+  type = PluginType.BEFORE as const;
+
+  // this.config is defined in setup() which will always be called first
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  config: NodeConfig;
+  eventId = 0;
+  library = `amplitude-ts/${VERSION}`;
+
+  setup(config: NodeConfig): Promise<undefined> {
+    this.config = config;
+    return Promise.resolve(undefined);
+  }
+
+  execute(context: Event): Promise<Event> {
+    return new Promise((resolve) => {
+      const time = new Date().getTime();
+
+      const contextEvent: Event = {
+        user_id: this.config.userId,
+        device_id: this.config.deviceId,
```

Comment:
i believe these values should come from client device since events processed by node sdk come from different user and device